### PR TITLE
Supplier category (#2088)

### PR DESCRIPTION
**Pull request by emi-silva**
This pull request corresponds to the second implementation step from Epic #2088.

### CODE REVIEW

**Comment by WebShells** [comment]
Neat and clean 👍

**Comment by jekkos** [comment]
@daN4cat I saw you set release milestone to 3.4. However if functionality is added in small chunks, we can easily start merging parts of it.

**Comment by emi-silva** [comment]
@SteveIreland you're totaly right. I'm gonna fix it by removing that default value and performing a data migration instead.

File: application/migrations/20180922162145_supplier_category.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	{
+		parent::__construct();
+	}
+
+	public function up()
+	{
+		$query	= "ALTER TABLE `ospos_suppliers` ";
+		$query .= "ADD COLUMN `category` VARCHAR(50) NOT NULL";
+		$this->db->query($query);
+		$category = $this->lang->line('suppliers_goods');
```
Comment (daN4cat) [inline application/migrations/20180922162145_supplier_category.php]: Please use a define, not a language dependent value.
There are a couple of existing cases language dependent and those are wrong and should be one day replaced.

**Comment by daN4cat** [comment]
Hmmm I don't particularly like the idea of using language dependent variable values in the database.
Can we switch to use defines and then replace those with language dependent strings in controller before being exposed to view or something down that line.

File: application/migrations/20180922162145_supplier_category.php (hunk 1 above)
Comment (SteveIreland) [reply application/migrations/20180922162145_supplier_category.php]: If the values are going to control program logic then I definitely agree with @daN4cat.

Plus, if I agree with him then he might go easier on my pull request.

**Comment by emi-silva** [comment]
OK, I'll change it. Any existing examples welcome.

**Comment by SteveIreland** [comment]
Although it is changing with the India GST project you can use the TAX_TYPE_xxx in Tax_lib.php as an example.

**Comment by emi-silva** [comment]
@daN4cat @SteveIreland Waiting for your approval

**Comment by jekkos** [comment]
Looks good to me now

**Comment by daN4cat** [comment]
@emi-silva have you locked your GitHUb repo or branch because I'm not able to pull the code for a merge.
I use the command line even because I need to add the en_GB translation before committing the change to master.

**Comment by emi-silva** [comment]
@daN4cat not at all

**Comment by emi-silva** [comment]
@daN4cat what error are you getting?

**Comment by daN4cat** [comment]
I cannot pull from you repo. I'll see if it was a temporary problem or something else.

**Comment by daN4cat** [comment]
git pull git://github.com/emi-silva/opensourcepos.git supplier-category
fatal: unable to connect to github.com:
github.com[0: 62.252.172.241]: errno=Connection timed out

**Comment by daN4cat** [comment]
It used to work, not sure what happened.... I need some time to check this story.

**Comment by emi-silva** [comment]
Maybe someone else can merge it? @jekkos?

**Comment by daN4cat** [comment]
I'll look into this today.

**Comment by daN4cat** [comment]
This is now in master.

Please note that I made some changes (e.g. Items form lists Goods supplier only, kept migration to current SQL script approach), and I squashed the commits.

**Pull request closed without merging**